File: src/core/tool-engine/runners/example-tool.py
"""
Example Python tool
This shows how to write a tool that runs in the sandboxed Python runner
"""

def run(args):
    """
    Tool function that processes arguments and returns result
    
    Args:
        args: Dictionary with tool arguments
        
    Returns:
        Dictionary with 'ok' (bool) and 'data' or 'error' fields
    """
    text = args.get("text", "")
    
    if not text:
        return {
            "ok": False,
            "error": {"message": "Missing required parameter: text"}
        }
    
    # Simple example: count words
    words = [w for w in text.strip().split() if w]
    
    return {
        "ok": True,
        "data": {
            "wordCount": len(words),
            "characterCount": len(text),
            "words": words[:10]  # First 10 words
        }
    }

# Alternative: use default name
# def default(args):
#     return run(args)

File: src/core/tool-engine/runners/example-tool.js
/**
 * Example JavaScript tool
 * This shows how to write a tool that runs in the sandboxed JS runner
 */

// Export default async function
export default async function run(args) {
  const { text } = args;
  
  if (!text) {
    return {
      ok: false,
      error: { message: "Missing required parameter: text" }
    };
  }
  
  // Simple example: count words
  const words = text.trim().split(/\s+/).filter(word => word.length > 0);
  
  return {
    ok: true,
    data: {
      wordCount: words.length,
      characterCount: text.length,
      words: words,
    }
  };
}

// Alternative: export named function
// export async function run(args) { ... }

// Alternative: CommonJS style
// module.exports = async function(args) { ... }

